modules/core/src/main/java/org/apache/fluo/core/oracle/OracleServer.java

```diff
@@ -357,8 +419,17 @@ public synchronized void stop() throws Exception {
       if (curatorFramework.getState().equals(CuratorFrameworkState.STARTED)) {
         pathChildrenCache.getListenable().removeListener(this);
         pathChildrenCache.close();
-        leaderSelector.close();
-        curatorFramework.getConnectionStateListenable().removeListener(this);
+        leaderLatch.close();
+
+        execService.shutdown();
+
+        execService.awaitTermination(10, TimeUnit.SECONDS);
+
+        curatorFramework.getConnectionStateListenable().removeListener(cnxnListener);
+
+        // leaderLatch.close() schedules a background delete, give it a chance to process before
+        // closing curator... this is done to avoid spurious exceptions, see CURATOR-467
+        Thread.sleep(250);
```

Comment:
An `InterruptedException` here will prevent the `curatorFramework.close()` on the next line. Maybe do something like the following?

```java
try {
  Thread.sleep(250);
} finally {
  curatorFramework.close();
}
```

Reply:
I can use the guava sleep uninterruptably